kong/cmd/utils/prefix_handler.lua

```diff
@@ -378,6 +381,67 @@ local function write_process_secrets_file(path, data)
   return true
 end
 
+local function pre_create_lmdb(conf)
+  local LMDB_DIRECTORY = kong_constants.LMDB_DIRECTORY
+
+  local prefix = conf.prefix
+  local lmdb_path = conf.lmdb_environment_path or LMDB_DIRECTORY
+
+  local dir_name
+  if lmdb_path:sub(1, 1) == "/" then
+    dir_name = lmdb_path
+
+  else
+    dir_name = prefix .. "/" .. lmdb_path
+  end
+
+  if pl_path.isdir(dir_name) then
+    return true
+  end
+
+  local user, group = match(conf.nginx_user or "", "(%w+)%s*(%w*)")
+
+  if not user then
+    log.warn("nginx_user is not set in Kong config, "..
+             "skipping verification of LMDB file permissions")
+    return true
+  end
+
+  if not group or #group == 0 then
+    group = user
+  end
+
+  log.debug("LMDB directory '%s' does not exist, " ..
+            "pre-creating with the correct permissions",
+            dir_name)
+
+  local ok, err = pl_path.mkdir(dir_name)
+  if not ok then
+    return nil, "can not create directory for LMDB " .. dir_name ..
+                ", err: " .. err
+  end
+
+  local cmds = {
+    format("chown %s:%s %s && chmod 0700 %s",
```

Comment:
The usage of `command && command` seems a little unnecessary given that the loop below fulfills the same purpose by exiting early after any command fails.